### PR TITLE
BCI-1426: Add GasPriceSubunits Data Source Pipeline 

### PR DESCRIPTION
**Pull request by augustbleeds**
Changes
* Add optional GasPriceSubunits pipeline in the plugin Config
* Validate the pipeline in ORM (if it exists) before insertion into postgres
* If gasPriceSubunits data source does not exist, create a zero data source. Pass it to the NewMedianFactory

Merge Order
1. https://github.com/smartcontractkit/chainlink-common/pull/509
2. https://github.com/smartcontractkit/chainlink-feeds/pull/28
3. This PR

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -2,7 +2,7 @@ golang 1.21.7
 mockery 2.42.2
 nodejs 20.13.1
 pnpm 8.15.8
-postgres 14.11
+postgres 15.1
```
Comment (augustbleeds) [inline .tool-versions:5]: Overdue for a bump?

File: plugins/medianpoc/plugin.go [hunk 2]
```diff
@@ -30,6 +30,14 @@ type Plugin struct {
 	reportingplugins.MedianProviderServer
 }
 
+type PipelineNotFoundError struct {
+	Message string
+}
+
+func (e *PipelineNotFoundError) Error() string {
+	return e.Message
+}
```
Comment (jmank88) [inline plugins/medianpoc/plugin.go]:
```suggestion
type PipelineNotFoundError struct {
	Key string
}

func (e *PipelineNotFoundError) Error() string {
	return fmt.Sprintf("no pipeline found for %s", e.Key)
}
```

File: core/services/ocr2/plugins/median/config/config.go [hunk 3]
```diff
@@ -41,5 +45,16 @@ func ValidatePluginConfig(config PluginConfig) error {
...
+	if !config.GasPriceSubunitsPipelineExists() {
+		return nil
+	} else if _, err := pipeline.Parse(config.GasPriceSubunitsPipeline); err != nil {
+		return errors.Wrap(err, "invalid gasPriceSubunitsSource pipeline")
+	}
+
 	return nil
 }
+
+func (config *PluginConfig) GasPriceSubunitsPipelineExists() bool {
```
Comment (jmank88) [inline core/services/ocr2/plugins/median/config/config.go]: What about something like `Has*() bool`?
```suggestion
func (config *PluginConfig) HasGasPriceSubunitsPipeline() bool {
```

File: core/services/ocr2/plugins/median/config/config.go [hunk 4]
```diff
@@ -41,5 +45,16 @@ func ValidatePluginConfig(config PluginConfig) error {
...
+	return !(strings.TrimSpace(config.GasPriceSubunitsPipeline) == "")
```
Comment (jmank88) [inline core/services/ocr2/plugins/median/config/config.go]:
```suggestion
	return strings.TrimSpace(config.GasPriceSubunitsPipeline) != ""
```

File: plugins/medianpoc/plugin.go [hunk 5]
```diff
@@ -103,12 +111,39 @@ func (p *Plugin) newFactory(ctx context.Context, config core.ReportingPluginServ
...
+
+	var gds median.DataSource
+	gp, err := jc.getPipeline("gasPriceSubunitsPipeline")
+
+	_, pipelineNotFound := err.(*PipelineNotFoundError)
+	if !pipelineNotFound && err != nil {
+		return nil, err
+	}
+
+	// Setting includeGasPriceSubunitsInObservation properly ensures we do not break libocr codec
```
Comment (krehermann) [inline plugins/medianpoc/plugin.go]: we had talked about updating this comment to be a more direct and simpler -- we have to omit the field in to maintain backward compatibility

**Comment by bolekk** [comment]
Is this ready to be merged? I'd like to update common dep to latest.

**Comment by augustbleeds** [comment]
> Is this ready to be merged? I'd like to update common dep to latest.

I'll bubble the PR up with the core team ~ it should be ready.

**Review by jmank88 — DISMISSED** [review]

**Review by bolekk — DISMISSED** [review]

**Comment by bolekk** [comment]
lint failure?

**Comment by augustbleeds** [comment]
> lint failure?

i *think* i fixed the lint. just pushed the fix

**Comment by augustbleeds** [comment]
@bolekk @jmank88 the linting error was fixed -- ready for another review (feel free to merge queue when it's ready)

**Review by bolekk — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

**Comment by augustbleeds** [comment]
there was a merge conflict so I had to resolve that. will need approvals again xD @jmank88 @bolekk

**Review by jmank88 — APPROVED** [review]

**Review by bolekk — APPROVED** [review]

**Comment by augustbleeds** [comment]
anyone able to add to merge queue?

**Pull request merged**